### DISCUSSION TITLE
QuartoPreview error - "attempt to index field 'config' (a nil  value)"

### DISCUSSION THREAD

**Discussion by deepakramani** [quarto-dev/quarto-nvim#57] [Q&A] [answered]
Hi there!
I'm getting a small error when executing `QuartoPreview` command. The error appears in the quickfix list window but the preview server opens up the preview content in the browser nonetheless. I have default setting for quarto and yet it errors. I don't know why this is happening. 

### Error
```
E5108: Error executing lua: ...ig/nvim/autoload/plugged/quarto-nvim/lua/quarto/init.lua:67: attempt to index field 'config' (a nil
 value)
stack traceback:
        ...ig/nvim/autoload/plugged/quarto-nvim/lua/quarto/init.lua:67: in function <...ig/nvim/autoload/plugged/quarto-nvim/lua/q
uarto/init.lua:29>
```

Using Macbook M1. 
```
$ nvim --version
NVIM v0.10.0-dev-109+ge83682e65-Homebrew
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/HEAD-e83682e/share/nvim"
```

With latest `vim-plug` I'm using the latest `otter`, `quarto-nvim`, `nvim-lspconfig`. 
```
silent! let g:plugs['otter.nvim'].commit = '5c5f1552223ba54e634aa2a61ba9a6004eaa0375'
silent! let g:plugs['quarto-nvim'].commit = 'e913e579e7531fc1c621900b481b8f49c0686ab4'
silent! let g:plugs['nvim-lspconfig'].commit = '591d5038552c6cdf6cfc595bc351ccdbb8e62045'
```

### quarto.lua
```
local quarto = require("quarto")
vim.keymap.set('n', '<leader>qp', quarto.quartoPreview, {silent = true, noremap = true})
```
I don't have a `otter.lua` file.

Can you please help? Thanks.

**Comment by jmbuhr** [comment, marked as answer]
In order to use all of quarto-nvims features you have to call `quarto.setup` (see https://github.com/quarto-dev/quarto-nvim#configure). Otherwise quarto.config is not set (https://github.com/quarto-dev/quarto-nvim/blob/9839eed2b7f2151721a0eb820e5a85a547201b63/lua/quarto/init.lua#L63).
But as it turns out I already had a fix for this, I just forgot to push it (thanks for reminding me)! If you update now you can use quarto preview without configuring quarto. Though I still recommend to call setup.

**Reply by deepakramani** [reply]
Thanks. Updated. Called `quarto.setup{}` and got it to work without errors.

Two more questions - 

1. Did you recently make changes to preview web server calls? I don't why but the `node` process is heating up my laptop. Battery level is reducing at a good rate(1-2% every 3-4 minutes). Didn't observe this behaviour two days ago?

2. I'm inside a directory that isn't yet mentioned in `index.qmd`. Right now `index.qmd` only lists `*.qmd` in the `/posts/` directory. So now that I'm in an another directory, I execute `QuartoPreview`. The preview window opens in the root `/` directory than the subdirectory location. The preview only opens as intended(subdirectory qmd) only when I make active changes to the file and preview again. Is this normal behaviour? In VSCode, I don't have such issues.

**Reply by jmbuhr** [reply]
1. Do you maybe have e.g. another language server running? `quarto` itself uses `deno`, not `node` and the `quarto-nvim` plugin does not directly talk to the preview server.
2. By root you mean the root of the quarto project, right? In this is case this is expected. After saving the file the preview server should then show you the recently saved file. This is because I currently don't directly interact with the preview server. It just knows what to show you based on watching file changes (i.e. saving).

**Reply by deepakramani** [reply]
>Do you maybe have e.g. another language server running? 
  
I installed [marksman](https://github.com/artempyanykh/marksman) yesterday. Other servers include pyright and texlab. It seems like `pyright` is triggering `node`. I'm sure the laptop didn't get this warm 5 days ago.  

**Update:**
Alright, I reverted `quarto-nvim` to [`0.7.3`](https://github.com/quarto-dev/quarto-nvim/releases/tag/v0.7.3). The process `node` doesn't run in the background anymore. I removed `marksman` as well. 
  
  >By root you mean the root of the quarto project, right? 
  
  Yes. 
  
  >It just knows what to show you based on watching file changes (i.e. saving).
  
  OK. I was wondering if this was something to do with my config. If I remember correctly, 2-3 days ago I could

**Reply by jmbuhr** [reply]
Nothing in quarto-nvim uses node, so reverting it shouldn't change this behavior.
Maybe a pyright update? I know pyright can be quite hungry.

**Reply by jmbuhr** [reply]
I'm assuming the fact that you reverted quarto-nvim but not otter.nvim and the neovim version just broke the language features, so pyright is not started anymore.

**Reply by deepakramani** [reply]
>Nothing in quarto-nvim uses node
>just broke the language features, so pyright is not started anymore.

Okay I understand that. Seems like my language servers aren't working as intended. I don't get code completion suggestions anymore. I have disabled `pyright` and back again in the latest quarto-nvim. I will debug pyright later. 

Thanks for your help.

**Reply by jmbuhr** [reply]
One thing to look out for might be the working directory that pyright thinks it should be working from. If e.g. it looks for a file that signifies a folder as the root of your project, but doesn't find it and then keeps looking upwards it may find one that is e.g. in your home folder. And then you have pyright thinking that your whole filesystem is one big python project.
`:LspInfo` will tell you.

**Reply by jmbuhr** [reply]
you can check the kickstarter config for an example.
